homeassistant/components/netatmo/light.py

```diff
@@ -0,0 +1,141 @@
+        """Initialize a Netatmo Presence camera light."""
+        self._camera_id = camera_id
+        self._data = camera_data
+        self._camera_type = self._data.camera_data.get_camera(camera_id).get("type")
+        self._name = (
+            f"{MANUFACTURER} {self._data.camera_data.get_camera(camera_id).get('name')}"
+        )
+        self._is_on = False
+        self._unique_id = f"{self._camera_id}-{self._camera_type}-light"
+        self._vpnurl = self._localurl = None
+        self._verify_ssl = True
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def device_info(self):
+        """Return the device info for the sensor."""
+        return {
+            "identifiers": {(DOMAIN, self._camera_id)},
+            "name": self._name,
+            "manufacturer": MANUFACTURER,
+            "model": self._camera_type,
+        }
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this light."""
+        return self._unique_id
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return 0
+
+    @property
+    def should_poll(self):
+        """Return if we should poll this device."""
+        return True
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return False if unable to access real state of the entity."""
+        return False
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._is_on
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        _LOGGER.debug("Set the flood light on for the camera '%s'", self._name)
+        if self._set_light_mode("on"):
+            self._data.camera_data.get_camera(self._camera_id)[
+                "light_mode_status"
+            ] = "on"
+        self.schedule_update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        _LOGGER.debug("Set the flood light off for the camera '%s'", self._name)
+        if self._set_light_mode("off"):
+            self._data.camera_data.get_camera(self._camera_id)[
+                "light_mode_status"
+            ] = "off"
+        self.schedule_update_ha_state()
+
+    def update(self):
+        """Update the camera data."""
+        self._data.update()
+        (self._vpnurl, self._localurl) = self._data.camera_data.camera_urls(
+            cid=self._camera_id
+        )
+        if self._data.camera_data.get_light_state(self._camera_id) == "on":
+            self._is_on = True
+        else:
+            self._is_on = False
+
+    def _set_light_mode(self, mode: str):
+        """Set light mode ('auto', 'on', 'off')."""
+        try:
+            camera_url = self._localurl if self._localurl else self._vpnurl
+            params = {"config": f'{{"mode":"{mode}"}}'}
+
+            resp = requests.get(
```

Comment:
I don't think we should keep this code within home assistant. Our guidelines are pretty clear about this.

I suggest we don't add this new platform and remove this feature now until it can be supported by a 3rd party library.

Reply:
I removed the code and will discuss to get this integrated in the upstream module.

Reply:
Good!